src/StreamingDataFrames/streamingdataframes/app.py

```diff
@@ -388,8 +380,8 @@ def run(
                     first_row.offset,
                 )
 
-                if self.is_stateful:
-                    # Store manager has stores registered, starting a transaction
+                if self._state_manager.stores:
```

Comment:
Can we do this check once before entering the run loop to determine which approach we should use?

Reply:
Yes, here we check if SDF requires any stores before processing. 
If there are no stores, it's a stateless SDF  and we don't need to manage any state partitions, etc.

Technically, you can do `if self._state_manager.stores:` check before the loop, but you still would need to have some `if` before starting each transaction because it requires topic-partition-offset

Reply:
Right, I was thinking about something like:

```
### before loop
if stores:
    t_maker = self._state_manager.start_store_transaction
else:
    t_maker = lambda *args: contextlib.nullcontext()

### in loop:
state_transaction = t_maker(t, p, o)
```

TBH not sure if a function call or if statement is faster though...probably an if statement, which would nullify my suggestion.

Reply:
Oh, I see. I like this idea.
It's indeed a bit faster because we can re-use the same `nullcontext` object many times.

I pushed the change